### PR TITLE
6807 google chat components

### PR DESCRIPTION
**Pull request by alysonturing**
### <samp>🤖 Generated by Copilot at 9a01993</samp>

This pull request adds a new app component and several action components for Google Chat, enabling users to interact with the Google Chat API from Pipedream. It also updates the `pnpm-lock.yaml` file and modifies the `package.json` file for the Google Chat components. It deletes some unused files from the previous version of the Google Chat app component.

### <samp>🤖 Generated by Copilot at 9a01993</samp>

> _This pull request adds some new features_
> _For Google Chat actions and triggers_
> _With props and app methods_
> _And `package.json` edits_
> _The components are now much richer_

### <samp>🤖 Generated by Copilot at 9a01993</samp>

*  Add a new app component for Google Chat that defines the authentication method, the prop definitions, and the methods for making requests to the Google Chat API ([link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-19092923e57b72ccbf476a9ac1cf0a9249c45521dc3c7cc3421542022673fe45R1-R156), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-11a21358d64230568cddb6e691720fa5a42ab40df685d0a206395d50e72a0d6cL5-R18), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1749-R1754))
* Add seven new action components for Google Chat that allow creating, getting, listing messages, members, and spaces ([link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-910d942b4e00eff6bb380ef7390dd99ad96d4a7308db938c4e648020cc618ca5R1-R32), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-98ee128de2f6075b5b43d5a485514d2727aa6c6119cac1be0cdc4ef5ed4ebda2R1-R36), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-30c06dd0153d5581b6a77dda6ad26ff136f0d98c5c1efd84c34cf67f1faffc84R1-R36), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-de62bfdb898649aa41768d3dea2fa2c767a944fc9d4cbd79ebe7ab76a5a76800R1-R26), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-9f7f683a0d6e5aacb662f70ea9d460ba6f0a4b30f51837842dabdffdbdfaee1aR1-R56), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-327cb62929e735de0854943aed81f3eefdbfb245e2e17dc2e5b1e182eb74152aR1-R67), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-f950ee474757ccebccf8854421ebe92ce9fc0d9f662f286d92db54c14e553c2bR1-R47))
* Delete unused files `components/google_chat/.gitignore` and `components/google_chat/app/google_chat.app.ts` ([link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-8202838b308d71136509090ac97762874a3b42e256ac0500b8e268698f61897a), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-bce307b5d3157a8e474b3df63dc926055a47c1160fdf21cef2d79d9aeff6d85b))
* Add new entries for other component packages to the `pnpm-lock.yaml` file, likely related to other pull requests ([link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR752-R754), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1393-R1395), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1492-R1494), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR1879-R1881), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2460-R2462), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2660-R2662), [link](https://github.com/PipedreamHQ/pipedream/pull/6867/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR4627-R4629))

### CODE REVIEW

File: components/google_chat/actions/list-spaces/list-spaces.mjs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    spaceType: {
+      type: "string[]",
+      label: "Space Type",
+      description: "You can filter spaces by the space type.",
+      optional: true,
+      options: [
+        "SPACE",
+        "GROUP_CHAT",
+        "DIRECT_MESSAGE",
+      ],
```
Comment (lcaresia) [inline components/google_chat/actions/list-spaces/list-spaces.mjs:32]: Move this to a constants.mjs file

```suggestion
      options: constants.SPACE_TYPES,
```

**Review by lcaresia — CHANGES_REQUESTED** [review]

**Review by lcaresia — APPROVED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Google_Chat_6867_2332.pdf](https://github.com/PipedreamHQ/pipedream/files/11772335/Google_Chat_6867_2332.pdf)

**Review by lcaresia — APPROVED** [review]

**Review by lcaresia — DISMISSED** [review]

**Review by lcaresia — APPROVED** [review]

**Pull request merged**